adobe_vipm/flows/sync/subscription.py

```diff
@@ -0,0 +1,172 @@
+import datetime as dt
+import logging
+import traceback
+
+from mpt_extension_sdk.mpt_http import mpt
+from mpt_extension_sdk.mpt_http.base import MPTClient
+
+from adobe_vipm.airtable import models
+from adobe_vipm.flows import utils as flows_utils
+from adobe_vipm.flows.constants import Param
+from adobe_vipm.flows.utils import notification
+from adobe_vipm.flows.utils.template import get_template_data_by_adobe_subscription
+from adobe_vipm.utils import get_commitment_start_date
+
+logger = logging.getLogger(__name__)
+
+
+class SubscriptionSyncer:
+    """
+    Handles synchronization of subscriptions between internal systems and Adobe.
+
+    Attributes:
+        mpt_client (MPTClient): The client used to interact with the MPT system.
+        agreement (dict): The agreement data that includes product and listing details.
+        customer (dict): The customer data including coterm date and other details.
+        adobe_subscriptions (list[dict]): A list of Adobe subscriptions to be synchronized.
+        subscriptions_for_update (list[dict]): A list of subscriptions requiring updates.
+        dry_run (bool): Specifies whether the operation should simulate updates without
+            making actual changes.
+    """
+
+    def __init__(
+        self,
+        mpt_client: MPTClient,
+        agreement: dict,
+        customer: dict,
+        adobe_subscriptions: list[dict],
+        subscriptions_for_update: list[dict],
+        *,
+        dry_run: bool,
+    ):
+        self._subscriptions_for_update = subscriptions_for_update
+        self._mpt_client = mpt_client
+        self._agreement = agreement
+        self._product_id = agreement["product"]["id"]
+        self._currency = agreement["listing"]["priceList"]["currency"]
+        self._agreement_id = agreement["id"]
+        self._authorization_id: str = agreement["authorization"]["id"]
+        self._seller_id: str = agreement["seller"]["id"]
+        self._licensee_id: str = agreement["licensee"]["id"]
```

Comment:
They should be properties to avoid inconsistencies and prevent changing them directly. We should update these properties through the agreement or if needed, by using the property.setter which should update the self._agreement as well